lib/daru/maths/statistics/vector.rb

```diff
@@ -855,6 +835,14 @@ def raw_sample_without_replacement sample
 
           out.collect { |i| valid[i] }
         end
+
+        def sort_without_nil(&block)
+          dv = reject_values(nil)
```

Comment:
TBH, it looks for me like `data = @data.data.to_a.compact` and then sorting it could be much more efficient.